### PR TITLE
Fix `CompileLoss` with undefined `loss` argument

### LINKED ISSUE #20243: support for dictionary-type loss functions without explicitly declaring `None`
In previous versions of Keras, when providing a loss function as a dictionary for models with multiple outputs, the compileLoss function would automatically assign a default value for outputs without a defined loss. The relevant code was:
```            for name, yt, yp in zip(output_names, y_true, y_pred):
                if name in loss:
                    if loss[name]:
                        flat_losses.append(get_loss(loss[name], yt, yp))
                    else:
                        flat_losses.append(None)
                else:
                    flat_losses.append(None)
``` 
 However, in the latest version, flat_losses is now simply:
  ``` flat_losses = tree.flatten(loss) ``` 
Is there a way to reintroduce support for dictionary-type loss functions without explicitly declaring None for undefined losses? This change reduces flexibility in using dictionary-based loss definitions.

### PR DESCRIPTION
**Pull request by james77777778**
Fix #20243 

Actually, #20243 works as expected in torch and jax backends without this PR. The underlying issue in TF backend has been resolved.

The root cause was that we used `pop` in `_filter_unused_inputs` within `CompileLoss`, which is not allowed in a jitted tf graph. The solution is to avoid modifying the incoming data.

A test has been added to verify the implementation.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you
